src/constant.h

```diff
@@ -7,594 +7,598 @@
+//constexpr int32_t SCARE_MONSTER = 99
 
 // Descriptive constants
-#define MAX_COLORS               49   // Used with potions
-#define MAX_MUSH                 22   // Used with mushrooms
-#define MAX_WOODS                25   // Used with staffs
-#define MAX_METALS               25   // Used with wands
-#define MAX_ROCKS                32   // Used with rings
-#define MAX_AMULETS              11   // Used with amulets
-#define MAX_TITLES               45   // Used with scrolls
-#define MAX_SYLLABLES           153   // Used with scrolls
+constexpr int32_t MAX_COLORS = 49; // Used with potions
+constexpr int32_t MAX_MUSH = 22; // Used with mushrooms
+constexpr int32_t MAX_WOODS = 25; // Used with staffs
+constexpr int32_t MAX_METALS = 25; // Used with wands
+constexpr int32_t MAX_ROCKS = 32; // Used with rings
+constexpr int32_t MAX_AMULETS = 11; // Used with amulets
+constexpr int32_t MAX_TITLES = 45; // Used with scrolls
+constexpr int32_t MAX_SYLLABLES = 153; // Used with scrolls
 
 // Player constants
-#define MAX_PLAYER_LEVEL         40   // Maximum possible character level
-#define MAX_EXP             9999999L  // Maximum amount of experience -CJS-
-#define MAX_RACES                 8   // Number of defined races
-#define MAX_CLASS                 6   // Number of defined classes
-#define USE_DEVICE                3   // x> Harder devices x< Easier devices
-#define MAX_BACKGROUND          128   // Number of types of histories for univ
-#define PLAYER_FOOD_FULL      10000   // Getting full
-#define PLAYER_FOOD_MAX       15000   // Maximum food value, beyond is wasted
-#define PLAYER_FOOD_FAINT       300   // Character begins fainting
-#define PLAYER_FOOD_WEAK       1000   // Warn player that he is getting very low
-#define PLAYER_FOOD_ALERT      2000   // Warn player that he is getting low
-#define PLAYER_REGEN_FAINT       33   // Regen factor*2^16 when fainting
-#define PLAYER_REGEN_WEAK        98   // Regen factor*2^16 when weak
-#define PLAYER_REGEN_NORMAL     197   // Regen factor*2^16 when full
-#define PLAYER_REGEN_HPBASE    1442   // Min amount hp regen*2^16
-#define PLAYER_REGEN_MNBASE     524   // Min amount mana regen*2^16
-#define PLAYER_WEIGHT_CAP       130   // "#"*(1/10 pounds) per strength point
-#define PLAYER_EXIT_PAUSE         2   // Pause time before player can re-roll
+constexpr int32_t MAX_PLAYER_LEVEL = 40; // Maximum possible character level
+constexpr int32_t MAX_EXP = 9999999L; // Maximum amount of experience -CJS-
+constexpr int32_t MAX_RACES = 8; // Number of defined races
+constexpr int32_t MAX_CLASS = 6; // Number of defined classes
+constexpr int32_t USE_DEVICE = 3; // x> Harder devices x< Easier devices
+constexpr int32_t MAX_BACKGROUND = 128; // Number of types of histories for univ
+constexpr int32_t PLAYER_FOOD_FULL = 10000; // Getting full
+constexpr int32_t PLAYER_FOOD_MAX = 15000; // Maximum food value, beyond is wasted
+constexpr int32_t PLAYER_FOOD_FAINT = 300; // Character begins faint32_ting
```

Comment:
More accidental search/replace errors `fainting` changed to `faint32_ting`. You should probably check the whole file for more of these. :)